src/main/java/com/coveros/selenified/element/check/azzert/AssertMatches.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    /**
+     * Verifies that the element's value matches the regular expression pattern
+     * provided. If the element isn't present or an input, this will constitute a failure,
+     * same as a mismatch. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param expectedPattern the expected input value of the element
+     */
+    public void value(String expectedPattern) {
+        String value = checkValue(expectedPattern, 0, 0);
+        if (value == null) {
+            String reason = "No element found";
+            if (getElement().is().present()) {
+                reason = "Element not input";
+            }
+            assertNotNull(reason, value);
+        }
+        assertTrue("Value Mismatch: value of '" + value + "' doesn't match pattern '" + expectedPattern + "'", value.matches(expectedPattern));
+    }
+
+    /**
+     * Verifies that the element's selected option matches the regular expression pattern
+     * provided. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedPattern the expected input text of the element
+     */
+    public void selectedOption(String expectedPattern) {
+        String selectedOption = checkSelectedOption(expectedPattern, 0, 0);
+        if (selectedOption == null) {
+            String reason = "No element found";
+            if (getElement().is().present()) {
+                reason = "Element not select";
+            }
+            assertNotNull(reason, selectedOption);
+        }
+        assertTrue("Selected Option Mismatch: option of '" + selectedOption + "' doesn't match pattern '" + expectedPattern + "'",
+                selectedOption.matches(expectedPattern));
+    }
+
+    /**
+     * Verifies that the element's selected value  matches the regular expression pattern
+     * provided. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedPattern the expected input value of the element
+     */
+    public void selectedValue(String expectedPattern) {
+        String selectedValue = checkSelectedValue(expectedPattern, 0, 0);
+        if (selectedValue == null) {
+            String reason = "No element found";
+            if (getElement().is().present()) {
+                reason = "Element not select";
+            }
+            assertNotNull(reason, selectedValue);
+        }
+        assertTrue("Selected Value Mismatch: value of '" + selectedValue + "' doesn't match pattern '" + expectedPattern + "'",
+                selectedValue.matches(expectedPattern));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "selectedValue" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)